source/Halibut.Tests/Support/TestCases/ClientAndServiceTestCasesBuilder.cs

```diff
@@ -39,21 +43,29 @@ public IEnumerable<ClientAndServiceTestCase> Build()
                             recommendedIterations = StandardIterationCount.ForServiceType(serviceConnectionType, clientServiceTestVersion);
                         }
 
-                        if (!forceClientProxyTypes.Any())
+                        foreach (var serviceAsyncHalibutFeatureTestCase in serviceAsyncHalibutFeatureTestCases)
                         {
-                            yield return new ClientAndServiceTestCase(serviceConnectionType, networkConditionTestCase, recommendedIterations, clientServiceTestVersion, null);
-                        }
-                        else
-                        {
-                            if (clientServiceTestVersion.IsPreviousClient())
+                            if (clientServiceTestVersion.IsPreviousService() && serviceAsyncHalibutFeatureTestCase == AsyncHalibutFeature.Enabled)
+                            {
+                                continue;
```

Comment:
I wonder if we should remove this if statement. The `LatestClientAndPreviousServiceVersionBuilder` will throw if this happens, and this will be a good signal to the developer that they configured it wrong.

Otherwise, it will just not appear when they think it should.

Reply:
This does make it easier to use the single attribute that gives you all the latest to previous combos. But it does come at this cost. This is similar to existing attributes which let you say pls test this and then they just don't test the thing that can't be tested in that way.

Maybe all of that can be improved.